python/flask/app-run-security-config.yaml

```diff
@@ -0,0 +1,53 @@
+
+# Flask check3: Flask application app.run best practices https://flask.palletsprojects.com/en/1.1.x/api/?highlight=run#flask.Flask.run
+rules:
+  # # . Rules for app.run() scope
+  - id: app_run_with_guard
+    pattern: |
+      if $X == '__main__':
+        app.run(...)
+    message: Running app.run(...) with guard is recommended
+    languages: [python]
+    severity: OK
+  - id: app_run_with_guard
+    pattern: |
+      if $X == '__main__':
+        ...
+        app.run(...)
+    message: Running app.run(...) with guard is recommended
+    languages: [python]
+    severity: OK
+  - id: app_run_with_guard_no_option
+    pattern: |
+      if $X == '__main__':
+        app.run()
+    message: Running app.run() with guard is recommended
+    languages: [python]
+    severity: OK
+  - id: app_run_with_guard_no_option
+    pattern: |
+      if $X == '__main__':
+        ...
+        app.run()
+    message: Running app.run() with guard is recommended
+    languages: [python]
+    severity: OK
+  # - id: app_run_within_func
+  #   pattern: |
+  #     def $X(...):
```

Comment:
I hope to support those patterns this week.

Reply:
I will cross test by bringing in latest in sgrep repo to the anlayzer and testing!

Reply:
Per https://app.circleci.com/jobs/github/returntocorp/sgrep/68, i will merge  this as is